frontend/src/metabase/parameters/components/widgets/ParameterFieldWidget/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -659,3 +755,54 @@ function renderValue({
     />
   );
 }
+
+type RemappedValueProps = {
+  value: string;
+  fields: (Field | null)[];
+  customOptions: ComboboxItem[];
+};
+
+function RemappedValue({ fields, value, customOptions }: RemappedValueProps) {
+  const matchedOption = customOptions.find((option) => option.value === value);
```

Comment:
Previously custom labels worked almost accidentally, and they broke in many cases. This change makes it clear how exactly we're remapped values in this case.